bin/importCampaignsFromSkadi.ts

```diff
@@ -0,0 +1,195 @@
+    query: /* sql */ `
+      WITH campaigns AS (
+      SELECT
+        id,
+        JSONExtractString(metadata, 'user_id') user_id,
+        JSONExtractString(metadata, 'post_id') post_id,
+        JSONExtractString(metadata, 'cancelled') cancelled,
+        active,
+        start_date,
+        end_date,
+        budget
+      FROM
+        skadi.campaigns_pg
+      WHERE
+        JSONExtractString(metadata, 'report_type') = 'post_boost'
+        AND user_id != ''
+        AND post_id != ''),
+          minmax AS (
+      SELECT
+        id,
+        min(start_date) OVER () AS min_date,
+        max(end_date) OVER () AS max_date
+      FROM
+        campaigns)
+      SELECT
+        campaigns.id AS campaign_id,
+        campaigns.user_id AS user_id,
+        campaigns.post_id AS post_id,
+        campaigns.start_date AS start_date,
+        campaigns.end_date AS end_date,
+        sum(aid.impressions) AS impressions,
+        sum(acd.clicks) AS clicks,
+        uniqIfMerge(aid.unique_users) AS unique_users,
+        campaigns.active AS active,
+        campaigns.cancelled AS cancelled,
+        campaigns.budget AS budget,
+        IF(csp.current_budget < 0, 0, csp.current_budget) AS current_budget
+      FROM
+        skadi.ad_impressions_daily aid FINAL
+      LEFT JOIN skadi.ad_clicks_daily acd FINAL ON
+        aid.campaign_id = acd.campaign_id
+      LEFT JOIN campaigns ON
+        aid.campaign_id = toString(campaigns.id)
+      LEFT JOIN minmax ON
+        aid.campaign_id = toString(minmax.id)
+      LEFT JOIN skadi.campaign_states_pg csp ON
+        aid.campaign_id = toString(csp.id)
+      WHERE
+        aid.event_date BETWEEN minmax.min_date AND minmax.max_date
+        AND aid.campaign_id IN (
+        SELECT
+          id
+        FROM
+          campaigns)
+      GROUP BY
+        campaign_id,
+        active,
+        start_date,
+        end_date,
+        user_id,
+        post_id,
+        minmax.min_date,
+        minmax.max_date,
+        budget,
+        current_budget,
+        cancelled;
+    `,
+    format: 'JSONEachRow',
+    query_params: queryParams,
+  });
+
+  const result = z
+    .array(skadiCampaignClickhouseSchema)
+    .safeParse(await response.json());
+
+  if (!result.success) {
+    logger.error(
+      {
+        schemaError: result.error.issues[0],
+      },
+      'Invalid campaign data',
+    );
+
+    throw new Error('Invalid campaign data');
+  }
+
+  const { data } = result;
+
+  const chunks: CampaignPost[][] = [];
+  const chunkSize = 500;
+
+  data.forEach((item) => {
+    if (chunks.length === 0 || chunks[chunks.length - 1].length === chunkSize) {
+      chunks.push([]);
+    }
+
+    chunks[chunks.length - 1].push({
+      id: item.campaign_id,
+      referenceId: item.post_id,
+      userId: item.user_id,
+      type: CampaignType.Post,
+      endedAt: item.end_date,
+      state: campaignStateFromSkadi({ item }),
+      flags: {
+        budget: usdToCores(item.budget),
```

Comment:
On our side, we store the `budget` as the full budget. We can maybe multiply this with the result of `getAbsoluteDifferenceInDays` for the start date and end date.